src/editor/editorinteractive.cc

```diff
@@ -651,13 +661,33 @@ void EditorInteractive::draw(RenderTarget& dst) {
 			}
 		}
 
-		// Draw build help.
-		if (buildhelp()) {
-			const auto* overlay =
-			   get_buildhelp_overlay(tools_->current().nodecaps_for_buildhelp(field.fcoords, ebase));
+                const Widelands::NodeCaps nodecaps = tools_->current().nodecaps_for_buildhelp(field.fcoords, ebase);
+                const Widelands::NodeCaps maxcaps = tools_->current().maxcaps_for_buildhelp(field.fcoords, ebase);
+		// Draw build help for maximum building spaces.
+		if ((draw_flags_ & kMaximumBuildhelp) != 0) {
+			const auto* overlay = get_buildhelp_overlay(maxcaps);
+
+                        const float opacity = nodecaps == maxcaps ? 1.0f : 0.6f;
+
+			if (overlay != nullptr) {
+				blit_field_overlay(&dst, field, overlay->pic, overlay->hotspot, scale, opacity);
+			}
+		}
```

Comment:
```suggestion
		// Draw build help for maximum building spaces.
		bool buildhelp_drawn = false;
		if ((draw_flags_ & kMaximumBuildhelp) != 0) {
			const auto* overlay = get_buildhelp_overlay(maxcaps);
                        const float opacity = nodecaps == maxcaps ? 1.0f : 0.6f;
			if (overlay != nullptr) {
				blit_field_overlay(&dst, field, overlay->pic, overlay->hotspot, scale, opacity);
				buildhelp_drawn = nodecaps == maxcaps
			}
		}
```

Reply:
For some reason the button to commit this became disabled.

![commit changes](https://user-images.githubusercontent.com/49523197/151699031-1bd43424-cd65-4f49-bb9f-e24461403a0f.png)

Reply:
Sorry, I think I blew this. @tothxa could you try to do this again, because now it won't even compile.

Reply:
Here it is as a patch: 
[maxcaps_patch.txt](https://github.com/widelands/widelands/files/7965890/maxcaps_patch.txt)
(I changed it to evaluate nodecaps==maxcaps only once)

Reply:
> Sorry, I think I blew this. @tothxa could you try to do this again, because now it won't even compile.

Don't worry, it's my fault. I shouldn't have made it as 2 suggestions, and it even came out in the wrong order.

Reply:
Thanks! It should be fixed now.